libosrmc/osrmc.cc

```diff
@@ -26,9 +26,32 @@ int osrmc_is_abi_compatible(void) { return osrmc_get_version() >> 16u == OSRMC_V
 /* API */
 
 struct osrmc_error final {
+  std::string code;
   std::string message;
 };
 
+static void osrmc_error_from_exception(const std::exception& e, osrmc_error_t* error) {
+  *error = new osrmc_error{"Exception", e.what()};
+}
+
+static void osrmc_error_from_json(osrm::json::Object* json, osrmc_error_t* error) try {
+  if (json == nullptr) {
+    *error = new osrmc_error{"Unknown", ""};
+  }
```

Comment:
I think we can `assert json != nullptr`. The object itself should always be there, otherwise the user is calling this API the wrong way. Check out how we use this API on the call sites, the result json is always an (potentially empty) object but should never be a nullptr.

Reply:
Great point! I've updated it to assert instead.

Thanks a lot for all the help! 🙂